jadx-core/src/main/java/jadx/core/codegen/ConditionGen.java

```diff
@@ -144,12 +144,18 @@ private void addNot(CodeWriter code, CondStack stack, IfCondition condition) thr
 	}
 
 	private void addAndOr(CodeWriter code, CondStack stack, IfCondition condition) throws CodegenException {
-		String mode = condition.getMode() == Mode.AND ? " && " : " || ";
+		Mode mode = condition.getMode();
+		String modeStr = mode == Mode.AND ? " && " : " || ";
 		Iterator<IfCondition> it = condition.getArgs().iterator();
 		while (it.hasNext()) {
-			wrap(code, stack, it.next());
+			IfCondition next = it.next();
+			if (next.getMode() == mode) {
+				add(code, stack, next);
+			} else {
+				wrap(code, stack, next);
+			}
```

Comment:
Can you check: is it possible to move this mode check inside `wrap` method because it already contains call: `boolean wrap = isWrapNeeded(cond);` or maybe inside `isWrapNeeded` method?

I prefer to localize all logic in code generators or better - made such optimizations in visitors before codegen by using flags (like `DONT_WRAP`).